Ical.Net/Evaluation/RecurrencePatternEvaluator.cs

```diff
@@ -102,20 +102,20 @@ private RecurrencePattern ProcessRecurrencePattern(CalDateTime referenceDate)
     /// For example, if the search start date (start) is Wed, Mar 23, 12:19PM, but the recurrence is Mon - Fri, 9:00AM - 5:00PM,
     /// the start dates returned should all be at 9:00AM, and not 12:19PM.
     /// </summary>
-    private IEnumerable<DateTime> GetDates(CalDateTime seed, CalDateTime? periodStart, CalDateTime? periodEnd, int maxCount, RecurrencePattern pattern,
+    private IEnumerable<CalDateTime> GetDates(CalDateTime seed, CalDateTime? periodStart, CalDateTime? periodEnd, int maxCount, RecurrencePattern pattern,
```

Comment:
Maybe check the performance impact of changing the internals of this class to CalDateTime. Could be measurable.